app/views/Title.scala

```diff
@@ -18,13 +18,16 @@ package views
 
 import play.api.i18n.Messages
 
-case class Title(title: String)
+case class Title(headingKey: String, sectionKey: Option[String] = None, headingArgs: Seq[String] = Seq.empty) {
 
-object Title {
-  def apply(headingKey: String, sectionKey: Option[String])(implicit messages: Messages): Title =
-    if (sectionKey.isEmpty) {
-      Title(messages("title.format", messages(headingKey), messages("service.name")))
-    } else {
-      Title(messages("title.withSection.format", messages(headingKey), messages(sectionKey.getOrElse("")), messages("service.name")))
+  def format(implicit messages: Messages): String = {
+    val heading = messages(headingKey, headingArgs: _*)
+    val service = messages("service.name")
+
+    sectionKey match {
+      case Some(section) => messages("title.withSection.format", heading, messages(section), service)
```

Comment:
Nested messages, messages and inside heading with messages, another messages, and service that contains messages. If we will cover it by tests which will check autmatically if the keys exists it will fail

Reply:
I'm not sure that in that case it'll make some problems, but we had a problems with nested messages

Reply:
So you're suggesting that we construct the Title with the translated message parts (i.e. heading and optional sub-heading)?  And get the Title to use those values as inputs to "title.withSection.format" etc?

I can do that if we get another vote for it

Reply:
You're right, it was a different case